proxy/src/compute.rs

```diff
@@ -282,12 +284,24 @@ impl ConnCfg {
         let (socket_addr, stream, host) = self.connect_raw(timeout).await?;
         drop(pause);
 
-        let tls_connector = native_tls::TlsConnector::builder()
-            .danger_accept_invalid_certs(allow_self_signed_compute)
-            .build()
-            .unwrap();
-        let mut mk_tls = postgres_native_tls::MakeTlsConnector::new(tls_connector);
-        let tls = MakeTlsConnect::<tokio::net::TcpStream>::make_tls_connect(&mut mk_tls, host)?;
+        let client_config = if allow_self_signed_compute {
```

Comment:
Can you add a comment explaining what this does. What exactly do we want to allow and what to deny here?

Reply:
IDK I just assumed that allowing everything is what is what the parameter controls. Maybe I'm wrong, waiting for a proxy expert to chime in. In the old version, I couldn't find the root store being populated by a whitelist of self signed certificates, so it's not just a "accept self signed certs in general but still check that it's from the allowed list" but a "accept self signed certs and anything else" option.

Reply:
We only use this feature for testing purposes. I've considered trying to remove it altogether. This seems fine to me

Reply:
Are you sure it's only used in tests? If I'm reading this correctly, `ConnCfg::connect` is called from `NodeInfo::connect`, which in turn is called from `TcpMechanism::connect_once`, which looks like production code to me.

If it is only used in tests, perhaps it could be moved to tests/ directory.

Reply:
Oh I just saw https://github.com/neondatabase/neon/pull/7884. You meant that the 'allow_self_signed_compute` flag is only used by tests, gotcha